cmd/kubeadm/app/cmd/upgrade/apply.go

```diff
@@ -72,11 +72,28 @@ func NewCmdApply(parentFlags *cmdUpgradeFlags) *cobra.Command {
 			err := runPreflightChecks(flags.parent.skipPreFlight)
 			kubeadmutil.CheckErr(err)
 
-			err = cmdutil.ValidateExactArgNumber(args, []string{"version"})
+			client, err := getClient(flags.parent.kubeConfigPath, false)
 			kubeadmutil.CheckErr(err)
+			// If the version is specified in config file, pick up that value.
+			if flags.parent.cfgPath != "" {
+				cfg, err := upgrade.FetchConfiguration(client, os.Stdout, flags.parent.cfgPath)
```

Comment:
Can you somehow split this out to only read the config file and determine whether it was set?
This is not correct I think, as it shouldn't ask the cluster for that information.
What we want to do here is read from the config only; and decide based on that.

Reply:
@luxas I rearranged a bit, PTAL